cabal-install/Distribution/Client/Setup.hs

```diff
@@ -2446,6 +2448,12 @@ userConfigCommand = CommandUI {
      "Overwrite the config file if it already exists."
      userConfigForce (\v flags -> flags { userConfigForce = v })
      trueArg
+ , option ['a'] ["append"]
```

Comment:
I'd call this `--append-lines`. Note that you'll still be able to say `--append=foo`, since `--append` is an unique prefix.